In [1]:
import numpy as np
import pandas as pd
import pylab as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

# Loading actor.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [2]:
data_actor = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\actor.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
data_actor.head()

actor_id first_name     last_name          last_update
0         1   PENELOPE       GUINESS  2006-02-15 04:34:33
1         2       NICK      WAHLBERG  2006-02-15 04:34:33
2         3         ED         CHASE  2006-02-15 04:34:33
3         4   JENNIFER         DAVIS  2006-02-15 04:34:33
4         5     JOHNNY  LOLLOBRIGIDA  2006-02-15 04:34:33

In [3]:
data_actor = data_actor.drop('last_update',axis=1)


In [4]:
data_actor.head()

actor_id first_name     last_name
0         1   PENELOPE       GUINESS
1         2       NICK      WAHLBERG
2         3         ED         CHASE
3         4   JENNIFER         DAVIS
4         5     JOHNNY  LOLLOBRIGIDA

In [5]:
cols1 = ['first_name','last_name']

In [6]:
data_actor[data_actor.duplicated(subset = cols1)]

actor_id first_name last_name
109       110      SUSAN     DAVIS

In [7]:
data_actor[(data_actor.first_name=='SUSAN') & (data_actor.last_name=='DAVIS')]

actor_id first_name last_name
100       101      SUSAN     DAVIS
109       110      SUSAN     DAVIS

### For the moment I am not going to drop this duplicate just in case there are really 2 Susan Davis

# Loading category.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [8]:
data_category = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\category.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)

data_category.head()

category_id       name          last_update
0            1     Action  2006-02-15 04:46:27
1            2  Animation  2006-02-15 04:46:27
2            3   Children  2006-02-15 04:46:27
3            4   Classics  2006-02-15 04:46:27
4            5     Comedy  2006-02-15 04:46:27

In [9]:
data_category = data_category.drop('last_update',axis=1)


In [10]:
data_category.head()

category_id       name
0            1     Action
1            2  Animation
2            3   Children
3            4   Classics
4            5     Comedy

In [11]:
cols1 = ['name']

In [12]:
data_category[data_category.duplicated(subset = cols1)]

Empty DataFrame
Columns: [category_id, name]
Index: []

#### There is no duplicates, so this table remains the same after dropping the 'last_update' column

# Loading film.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [13]:
data_film = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\film.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)

data_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  original_language_id  rental_duration  rental_rate  length  \
0            1                   NaN                6         0.99      86   
1            1                   NaN                3         4.99      48   
2            1                   NaN                7         2.99      50   
3            1                   NaN                5         2.99     117   
4            1                   NaN                6         2.99     130   

   replacement_cost rating                  special_features  \
0             20.99     PG  Deleted Scenes,Behind the Scenes   
1             12.99      G           Trailers,Deleted Scenes   
2             18.99  NC-17           Trailers,Deleted Scenes   
3             26.99      G    Commentaries,Behind the Scenes   
4             22.99      G                    Deleted Scenes   

           last_update  
0  2006-02-15 05:03:42  
1  2006-02-15 05:03:42  
2  2006-02-15 05:03:42  
3  2006-02-15 05:03:42  
4  2006-02-15 05:03:42

In [14]:
data_film = data_film.drop('last_update',axis=1)


In [15]:
data_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  release_year  \
0  A Epic Drama of a Feminist And a Mad Scientist...          2006   
1  A Astounding Epistle of a Database Administrat...          2006   
2  A Astounding Reflection of a Lumberjack And a ...          2006   
3  A Fanciful Documentary of a Frisbee And a Lumb...          2006   
4  A Fast-Paced Documentary of a Pastry Chef And ...          2006   

   language_id  original_language_id  rental_duration  rental_rate  length  \
0            1                   NaN                6         0.99      86   
1            1                   NaN                3         4.99      48   
2            1                   NaN                7         2.99      50   
3            1                   NaN                5         2.99     117   
4            1                   NaN                6         2.99     130   

   replacement_cost rating                  special_features  
0             20.99     PG  Deleted Scenes,Behind the Scenes  
1             12.99      G           Trailers,Deleted Scenes  
2             18.99  NC-17           Trailers,Deleted Scenes  
3             26.99      G    Commentaries,Behind the Scenes  
4             22.99      G                    Deleted Scenes

In [16]:
data_film.columns

Index(['film_id', 'title', 'description', 'release_year', 'language_id',
       'original_language_id', 'rental_duration', 'rental_rate', 'length',
       'replacement_cost', 'rating', 'special_features'],
      dtype='object')

In [17]:
cols1 = ['title', 'description', 'release_year', 'language_id',
       'original_language_id', 'rental_duration', 'rental_rate', 'length',
       'replacement_cost', 'rating', 'special_features']

cols2 = ['title']

In [18]:
data_film[data_film.duplicated(subset = cols1)]

Empty DataFrame
Columns: [film_id, title, description, release_year, language_id, original_language_id, rental_duration, rental_rate, length, replacement_cost, rating, special_features]
Index: []

In [19]:
data_film[data_film.duplicated(subset = cols2)]

Empty DataFrame
Columns: [film_id, title, description, release_year, language_id, original_language_id, rental_duration, rental_rate, length, replacement_cost, rating, special_features]
Index: []

#### There is no duplicates, so this table remains the same after dropping the 'last_update' column

In [20]:
data_film.shape

(1000, 12)

In [21]:
nan_cols = data_film.isna().sum()  # serie con los valores nulos por columna en porcentaj

nan_cols[nan_cols>0]


original_language_id    1000
dtype: int64

In [22]:
data_film.release_year.unique()

array([2006], dtype=int64)

#### There is no relevant data in original_language_id as the number of Nulls is equal as the lenght of the series, so I will drop this column. I will do the same to release_year, as all is full of 2006 and so it does not give us relevant information. I will keep language_id even though it is full of '1's but maybe in the future it could be filled up with useful information.

In [23]:
data_film = data_film.drop(['original_language_id','release_year'],axis=1)


In [24]:
data_film.head()

film_id             title  \
0        1  ACADEMY DINOSAUR   
1        2    ACE GOLDFINGER   
2        3  ADAPTATION HOLES   
3        4  AFFAIR PREJUDICE   
4        5       AFRICAN EGG   

                                         description  language_id  \
0  A Epic Drama of a Feminist And a Mad Scientist...            1   
1  A Astounding Epistle of a Database Administrat...            1   
2  A Astounding Reflection of a Lumberjack And a ...            1   
3  A Fanciful Documentary of a Frisbee And a Lumb...            1   
4  A Fast-Paced Documentary of a Pastry Chef And ...            1   

   rental_duration  rental_rate  length  replacement_cost rating  \
0                6         0.99      86             20.99     PG   
1                3         4.99      48             12.99      G   
2                7         2.99      50             18.99  NC-17   
3                5         2.99     117             26.99      G   
4                6         2.99     130             22.99      G   

                   special_features  
0  Deleted Scenes,Behind the Scenes  
1           Trailers,Deleted Scenes  
2           Trailers,Deleted Scenes  
3    Commentaries,Behind the Scenes  
4                    Deleted Scenes

In [25]:
data_film.length.unique()

array([ 86,  48,  50, 117, 130, 169,  62,  54, 114,  63, 126, 136, 150,
        94,  46, 180,  82,  57, 113,  79, 129,  85,  92, 181,  74, 179,
        91, 168, 119, 153, 147, 127, 121,  68,  99, 148, 137, 170,  83,
       108, 118, 162, 182,  75, 173,  87,  65,  90, 122, 160,  89, 175,
       106,  73, 151, 100,  53,  77, 142,  93, 163, 103,  61,  71, 102,
        76,  98,  72, 176, 123,  56, 161,  60, 133, 125,  67,  52, 120,
       167,  70, 135, 112, 183, 110, 152,  51, 146,  66, 185, 124, 107,
       101, 132, 143, 165,  81, 164,  58,  95, 139,  55, 149, 109,  59,
       172, 115, 184, 166,  64, 104,  69,  84, 144,  88, 141,  47,  49,
       177, 154, 159, 178,  96, 116, 171, 140, 155, 158, 174, 138,  97,
       131, 156,  80, 145, 111, 128, 157,  78, 105, 134], dtype=int64)

In [26]:
data_film.language_id.unique()

array([1], dtype=int64)

In [27]:
data_film.rating.unique()

array(['PG', 'G', 'NC-17', 'PG-13', 'R'], dtype=object)

In [28]:
data_film.special_features.unique()

array(['Deleted Scenes,Behind the Scenes', 'Trailers,Deleted Scenes',
       'Commentaries,Behind the Scenes', 'Deleted Scenes', 'Trailers',
       'Commentaries,Deleted Scenes',
       'Trailers,Deleted Scenes,Behind the Scenes',
       'Trailers,Commentaries,Behind the Scenes', 'Trailers,Commentaries',
       'Trailers,Behind the Scenes',
       'Commentaries,Deleted Scenes,Behind the Scenes',
       'Trailers,Commentaries,Deleted Scenes',
       'Trailers,Commentaries,Deleted Scenes,Behind the Scenes',
       'Behind the Scenes', 'Commentaries'], dtype=object)

# Loading inventory.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [29]:
data_inventory = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\inventory.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)

data_inventory.head()

inventory_id  film_id  store_id          last_update
0             1        1         1  2006-02-15 05:09:17
1             2        1         1  2006-02-15 05:09:17
2             3        1         1  2006-02-15 05:09:17
3             4        1         1  2006-02-15 05:09:17
4             5        1         2  2006-02-15 05:09:17

In [30]:
data_inventory = data_inventory.drop('last_update',axis=1)


In [31]:
data_inventory.head()

inventory_id  film_id  store_id
0             1        1         1
1             2        1         1
2             3        1         1
3             4        1         1
4             5        1         2

In [32]:
data_inventory.columns

Index(['inventory_id', 'film_id', 'store_id'], dtype='object')

In [33]:
cols1 = ['inventory_id', 'film_id', 'store_id']

In [34]:
data_inventory[data_inventory.duplicated(subset = cols1)]

Empty DataFrame
Columns: [inventory_id, film_id, store_id]
Index: []

#### No duplicates, so after dropping 'last_update' column this Dataframe remains the same

In [35]:
data_inventory.tail(5)

inventory_id  film_id  store_id
995           996      222         2
996           997      222         2
997           998      222         2
998           999      223         2
999          1000      223         2

In [36]:
data_inventory.describe().T

count     mean         std  min     25%    50%     75%     max
inventory_id  1000.0  500.500  288.819436  1.0  250.75  500.5  750.25  1000.0
film_id       1000.0  109.866   63.862042  1.0   56.00  111.5  164.00   223.0
store_id      1000.0    1.497    0.500241  1.0    1.00    1.0    2.00     2.0

In [37]:
inventory_id_lst = data_inventory['inventory_id']
inventory_id_lst = list(inventory_id_lst)

film_id_lst = data_inventory['film_id']
film_id_lst = list(film_id_lst)

store_id_lst = data_inventory['store_id']
store_id_lst = list(store_id_lst)

list

### Loop to create registers up to 4600 inventory_id by assigning to ascending order of film_id

In [40]:
inventory_id = 1000
film_id = 223


while inventory_id <4600:
    store_id = 0
    film_id += 1
    for i in range(2):
        store_id+=1
        for j in range(3):
            inventory_id += 1
            
            inventory_id_lst.append(inventory_id)
            film_id_lst.append(film_id)
            store_id_lst.append(store_id)




In [41]:
data_inventory = pd.DataFrame(list(zip(inventory_id_lst, film_id_lst,store_id_lst)),columns =['inventory_id', 'film_id', 'store_id'])

In [92]:
data_inventory.tail(25)

inventory_id  film_id  store_id
4575          4576      819         2
4576          4577      820         1
4577          4578      820         1
4578          4579      820         1
4579          4580      820         2
4580          4581      820         2
4581          4582      820         2
4582          4583      821         1
4583          4584      821         1
4584          4585      821         1
4585          4586      821         2
4586          4587      821         2
4587          4588      821         2
4588          4589      822         1
4589          4590      822         1
4590          4591      822         1
4591          4592      822         2
4592          4593      822         2
4593          4594      822         2
4594          4595      823         1
4595          4596      823         1
4596          4597      823         1
4597          4598      823         2
4598          4599      823         2
4599          4600      823         2

# Loading language.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [43]:
data_language = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\language.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)



In [44]:
data_language.head()

language_id      name          last_update
0            1   English  2006-02-15 05:02:19
1            2   Italian  2006-02-15 05:02:19
2            3  Japanese  2006-02-15 05:02:19
3            4  Mandarin  2006-02-15 05:02:19
4            5    French  2006-02-15 05:02:19

In [45]:
data_language = data_language.drop('last_update',axis=1)


In [46]:
data_language.head()

language_id      name
0            1   English
1            2   Italian
2            3  Japanese
3            4  Mandarin
4            5    French

In [47]:
data_language.columns

Index(['language_id', 'name'], dtype='object')

In [48]:
cols1 = ['name']

In [49]:
data_language[data_language.duplicated(subset = cols1)]

Empty DataFrame
Columns: [language_id, name]
Index: []

#### No duplicates, so after dropping 'last_update' column this Dataframe remains the same

# Loading rental.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates and dropping last_update column as it does not give us important information

In [50]:
data_rental = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\rental.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)



In [51]:
data_rental.head()

rental_id          rental_date  inventory_id  customer_id  \
0          1  2005-05-24 22:53:30           367          130   
1          2  2005-05-24 22:54:33          1525          459   
2          3  2005-05-24 23:03:39          1711          408   
3          4  2005-05-24 23:04:41          2452          333   
4          5  2005-05-24 23:05:21          2079          222   

           return_date  staff_id          last_update  
0  2005-05-26 22:04:30         1  2006-02-15 21:30:53  
1  2005-05-28 19:40:33         1  2006-02-15 21:30:53  
2  2005-06-01 22:12:39         1  2006-02-15 21:30:53  
3  2005-06-03 01:43:41         2  2006-02-15 21:30:53  
4  2005-06-02 04:33:21         1  2006-02-15 21:30:53

In [52]:
data_rental = data_rental.drop('last_update',axis=1)
data_rental.head()

rental_id          rental_date  inventory_id  customer_id  \
0          1  2005-05-24 22:53:30           367          130   
1          2  2005-05-24 22:54:33          1525          459   
2          3  2005-05-24 23:03:39          1711          408   
3          4  2005-05-24 23:04:41          2452          333   
4          5  2005-05-24 23:05:21          2079          222   

           return_date  staff_id  
0  2005-05-26 22:04:30         1  
1  2005-05-28 19:40:33         1  
2  2005-06-01 22:12:39         1  
3  2005-06-03 01:43:41         2  
4  2005-06-02 04:33:21         1

In [53]:
data_rental.columns

Index(['rental_id', 'rental_date', 'inventory_id', 'customer_id',
       'return_date', 'staff_id'],
      dtype='object')

In [54]:
cols2 = ['rental_id', 'rental_date','return_date', 'inventory_id', 'customer_id',
        'staff_id']

In [55]:
data_rental = data_rental[cols2]

In [56]:
data_rental.head()

rental_id          rental_date          return_date  inventory_id  \
0          1  2005-05-24 22:53:30  2005-05-26 22:04:30           367   
1          2  2005-05-24 22:54:33  2005-05-28 19:40:33          1525   
2          3  2005-05-24 23:03:39  2005-06-01 22:12:39          1711   
3          4  2005-05-24 23:04:41  2005-06-03 01:43:41          2452   
4          5  2005-05-24 23:05:21  2005-06-02 04:33:21          2079   

   customer_id  staff_id  
0          130         1  
1          459         1  
2          408         1  
3          333         2  
4          222         1

In [57]:
rental_date = []
return_date = []
total_time = []
for i in range(len(data_rental)):
    
    year_ren = data_rental.loc[i]['rental_date'][0:4]
    month_ren = data_rental.loc[i]['rental_date'][5:7]
    day_ren = data_rental.loc[i]['rental_date'][8:10]
    year_ret = data_rental.loc[i]['return_date'][0:4]
    month_ret = data_rental.loc[i]['return_date'][5:7]
    day_ret = data_rental.loc[i]['return_date'][8:10]
    
    rental_date.append(year_ren+'-'+month_ren+'-'+day_ren)
    return_date.append(year_ret+'-'+month_ret+'-'+day_ret)
    
data_rental['rental_date'] = rental_date    
    
data_rental['return_date'] = return_date    
    

In [58]:

data_rental['rental_date']= pd.to_datetime(data_rental['rental_date'])
data_rental['return_date']= pd.to_datetime(data_rental['return_date'])
data_rental.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   rental_id     1000 non-null   int64         
 1   rental_date   1000 non-null   datetime64[ns]
 2   return_date   1000 non-null   datetime64[ns]
 3   inventory_id  1000 non-null   int64         
 4   customer_id   1000 non-null   int64         
 5   staff_id      1000 non-null   int64         
dtypes: datetime64[ns](2), int64(4)
memory usage: 47.0 KB


In [59]:
import datetime
from dateutil.relativedelta import relativedelta
from datetime import date

In [60]:
date1 = data_rental['rental_date']
date2 = data_rental['return_date']

In [61]:
date_df = pd.DataFrame(dict(Start_date = date1, End_date = date2))

date_df.head()

Start_date   End_date
0 2005-05-24 2005-05-26
1 2005-05-24 2005-05-28
2 2005-05-24 2005-06-01
3 2005-05-24 2005-06-03
4 2005-05-24 2005-06-02

In [62]:
date_df['diff_days'] = date_df['End_date'] - date_df['Start_date']

date_df.head()

Start_date   End_date diff_days
0 2005-05-24 2005-05-26    2 days
1 2005-05-24 2005-05-28    4 days
2 2005-05-24 2005-06-01    8 days
3 2005-05-24 2005-06-03   10 days
4 2005-05-24 2005-06-02    9 days

In [63]:
data_rental['rental_time'] = date_df['diff_days']

In [64]:
data_rental.head()

rental_id rental_date return_date  inventory_id  customer_id  staff_id  \
0          1  2005-05-24  2005-05-26           367          130         1   
1          2  2005-05-24  2005-05-28          1525          459         1   
2          3  2005-05-24  2005-06-01          1711          408         1   
3          4  2005-05-24  2005-06-03          2452          333         2   
4          5  2005-05-24  2005-06-02          2079          222         1   

  rental_time  
0      2 days  
1      4 days  
2      8 days  
3     10 days  
4      9 days

In [65]:
data_rental.columns

Index(['rental_id', 'rental_date', 'return_date', 'inventory_id',
       'customer_id', 'staff_id', 'rental_time'],
      dtype='object')

In [66]:
cols_rental = ['rental_id', 'rental_date', 'return_date','rental_time', 'inventory_id',
       'customer_id', 'staff_id']

In [67]:
data_rental = data_rental[cols_rental]

In [68]:
data_rental.head()

rental_id rental_date return_date rental_time  inventory_id  customer_id  \
0          1  2005-05-24  2005-05-26      2 days           367          130   
1          2  2005-05-24  2005-05-28      4 days          1525          459   
2          3  2005-05-24  2005-06-01      8 days          1711          408   
3          4  2005-05-24  2005-06-03     10 days          2452          333   
4          5  2005-05-24  2005-06-02      9 days          2079          222   

   staff_id  
0         1  
1         1  
2         1  
3         2  
4         1

In [69]:
data_rental['rental_time'].unique()

array([172800000000000, 345600000000000, 691200000000000, 864000000000000,
       777600000000000, 259200000000000, 432000000000000, 518400000000000,
        86400000000000,               0, 604800000000000],
      dtype='timedelta64[ns]')

In [70]:
data_rental.loc[0]['rental_time']

Timedelta('2 days 00:00:00')

### I need to change the rental time column to a integer so I can compare with the rental_duration of films table

In [71]:
data_rental['rental_time'] = pd.to_numeric(data_rental['rental_time'].dt.days, downcast='integer')

In [72]:
data_rental.head()

rental_id rental_date return_date  rental_time  inventory_id  customer_id  \
0          1  2005-05-24  2005-05-26            2           367          130   
1          2  2005-05-24  2005-05-28            4          1525          459   
2          3  2005-05-24  2005-06-01            8          1711          408   
3          4  2005-05-24  2005-06-03           10          2452          333   
4          5  2005-05-24  2005-06-02            9          2079          222   

   staff_id  
0         1  
1         1  
2         1  
3         2  
4         1

In [73]:
inventory_id = data_rental['inventory_id']

In [74]:
inventory_id.shape

(1000,)

In [75]:
inventory_id.sort_values().tail()

675    4558
871    4568
869    4573
580    4579
709    4581
Name: inventory_id, dtype: int64

### From this series we can observe that inventory_id goes up to 4581 and on the other hand on the inventory table it goes only up to 1000 while reaching film_id 225. So we are going to create new registers in the inventory table...so they can match

# Loading old_HDD.csv to start cleaning. 

#### Cleaning process in this case will imply checking duplicates 

In [76]:
data_old_HDD = pd.read_csv(r'C:\Users\pombo\OneDrive\Escritorio\Ironhack\Labs\Day15Proy\w3-database-project\data\old_HDD.csv', encoding='latin1' )
pd.set_option('display.max_columns', None)

data_old_HDD.head()

first_name last_name                  title  release_year  category_id
0   PENELOPE   GUINESS       ACADEMY DINOSAUR          2006            6
1   PENELOPE   GUINESS   ANACONDA CONFESSIONS          2006            2
2   PENELOPE   GUINESS            ANGELS LIFE          2006           13
3   PENELOPE   GUINESS  BULWORTH COMMANDMENTS          2006           10
4   PENELOPE   GUINESS          CHEAPER CLYDE          2006           14

In [77]:
data_old_HDD.head(10)

first_name last_name                  title  release_year  category_id
0   PENELOPE   GUINESS       ACADEMY DINOSAUR          2006            6
1   PENELOPE   GUINESS   ANACONDA CONFESSIONS          2006            2
2   PENELOPE   GUINESS            ANGELS LIFE          2006           13
3   PENELOPE   GUINESS  BULWORTH COMMANDMENTS          2006           10
4   PENELOPE   GUINESS          CHEAPER CLYDE          2006           14
5   PENELOPE   GUINESS     COLOR PHILADELPHIA          2006            4
6   PENELOPE   GUINESS        ELEPHANT TROJAN          2006           11
7   PENELOPE   GUINESS    GLEAMING JAWBREAKER          2006           15
8   PENELOPE   GUINESS         HUMAN GRAFFITI          2006           10
9   PENELOPE   GUINESS         KING EVOLUTION          2006            8

In [78]:
### Clean old_HDD data:

actor_id = []

for i in range(len(data_old_HDD)):

    for j in range(len(data_actor)):

        if data_old_HDD.loc[i]['first_name'] == data_actor.loc[j]['first_name'] and data_old_HDD.loc[i]['last_name'] == data_actor.loc[j]['last_name']:
            actor_id.append(data_actor.loc[j]['actor_id'])
            break
            
data_old_HDD['actor_id'] = actor_id

film_id = []

for i in range(len(data_old_HDD)):
    
    for j in range(len(data_film)):

        if data_old_HDD.loc[i]['title'] == data_film.loc[j]['title']:
            film_id.append(data_film.loc[j]['film_id'])
            break

data_old_HDD['film_id'] = film_id

data_old_HDD = data_old_HDD.drop('first_name',axis=1)
data_old_HDD = data_old_HDD.drop('last_name',axis=1)
data_old_HDD = data_old_HDD.drop('title',axis=1)
data_old_HDD = data_old_HDD.drop('release_year',axis=1)

In [79]:
cols3 = ['actor_id', 'film_id', 'category_id']

data_old_HDD = data_old_HDD[cols3]

In [80]:
data_old_HDD.head()

actor_id  film_id  category_id
0         1        1            6
1         1       23            2
2         1       25           13
3         1      106           10
4         1      140           14

In [81]:
data_old_HDD.head()

actor_id  film_id  category_id
0         1        1            6
1         1       23            2
2         1       25           13
3         1      106           10
4         1      140           14

# SQL

In [82]:
%pip install mysql-connector-python
%pip install SQLAlchemy
%pip install pymysql

Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.
Note: you may need to restart the kernel to use updated packages.


In [83]:
from sqlalchemy import create_engine
import mysql.connector as conn

In [84]:
# crear base de datos nueva
from pass_ import *


# crear una base de datos

# string de conexion URI
# lenguaje+driver://usuario:password@servidor:puerto

str_conn='mysql+pymysql://root:password@localhost:3306'

cursor=create_engine(str_conn)


cursor.execute('drop database if exists videoclub;')

cursor.execute('create database videoclub;')

str_conn='mysql+pymysql://root:password@localhost:3306/videoclub'
cursor=create_engine(str_conn)

'''

conexion=conn.connect(host='localhost', user='root', passwd=pass_) # conexion al servidor

cursor=conexion.cursor()

cursor.execute('drop database if exists videoclub;')

cursor.execute('create database videoclub;')

'''

"\n\nconexion=conn.connect(host='localhost', user='root', passwd=pass_) # conexion al servidor\n\ncursor=conexion.cursor()\n\ncursor.execute('drop database if exists videoclub;')\n\ncursor.execute('create database videoclub;')\n\n"

In [85]:
data_actor.to_sql(name='actors', con=cursor, if_exists='replace', index=False)

200

In [86]:
data_category.to_sql(name='category', con=cursor, if_exists='replace', index=False)

16

In [87]:
data_film.to_sql(name='film', con=cursor, if_exists='replace', index=False)

1000

In [88]:
data_inventory.to_sql(name='inventory', con=cursor, if_exists='replace', index=False)

4600

In [89]:
data_language.to_sql(name='language', con=cursor, if_exists='replace', index=False)

6

In [90]:
data_rental.to_sql(name='rental', con=cursor, if_exists='replace', index=False)

1000

In [91]:
data_old_HDD.to_sql(name='old_hdd', con=cursor, if_exists='replace', index=False)

1000